homeassistant/components/freebox/switch.py

```diff
@@ -18,49 +20,67 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class FreeboxSwitchRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    state_fn: Callable[[FreeboxRouter], Any]
+    on_off_fn: Callable[[FreeboxRouter], Any]
+
+
+@dataclass
+class FreeboxSwitchEntityDescription(
+    SwitchEntityDescription, FreeboxSwitchRequiredKeysMixin
+):
+    """Describes Freebox switch entity."""
+
+
+SWITCH_DESCRIPTIONS = [
+    FreeboxSwitchEntityDescription(
+        key="wifi",
+        name="Freebox WiFi",
+        entity_category=EntityCategory.CONFIG,
+        state_fn=lambda router: router.wifi.get_global_config,
+        on_off_fn=lambda router: router.wifi.set_global_config,
+    )
+]
+
+
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the switch."""
     router: FreeboxRouter = hass.data[DOMAIN][entry.unique_id]
-    async_add_entities([FreeboxWifiSwitch(router)], True)
-
-
-class FreeboxWifiSwitch(SwitchEntity):
-    """Representation of a freebox wifi switch."""
-
-    def __init__(self, router: FreeboxRouter) -> None:
-        """Initialize the Wifi switch."""
-        self._name = "Freebox WiFi"
-        self._state: bool | None = None
-        self._router = router
-        self._unique_id = f"{self._router.mac} {self._name}"
+    async_add_entities(
+        [
+            FreeboxSwitch(
+                router,
+                entity_description,
+            )
+            for entity_description in SWITCH_DESCRIPTIONS
+        ],
+        True,
+    )
```

Comment:
Small style adjustment, I think this looks prettier:
```suggestion
    entities = [
        FreeboxSwitch(router, entity_description)
        for entity_description in SWITCH_DESCRIPTIONS
    ]
    async_add_entities(entities, True)
```